149: Add tests for api/posts endpoints

EugeneTurkin opened the pull request:
We want to cover our REST API part of the codebase with unit-tests.

In the scope of this task we need to cover code which is working with posts in our REST API.

birthdaysgift commented on tests/posts/test_posts_list_endpoint.py:
```diff
@@ -0,0 +1,266 @@
+    assert result.status_code == 200
+    response_body = result.json
+    post_id = response_body.pop("id")
+    assert isinstance(post_id, int)
```
Here we need to also check `created_at` the same way we check `post_id`:
```python
created_at = response_body.pop("created_at")
assert datetime.strptime(created_at, "%Y-%m-%d %H:%M:%S.%f") < datetime.utcnow()
assert response_body == {
    # "id": 10000,  # noqa
    "author_id": 1,
    "body": "test",
    # "created_at": "2023-05-07 11:25:17.592851",
    "topic_id": 1,
}
```

birthdaysgift commented on tests/posts/test_posts_list_endpoint.py:
```diff
@@ -0,0 +1,266 @@
+    assert isinstance(created_post.id, int)
+
+
+def test_post_posts_list_endpoint_with_nonexistent_topic_returns_404_with_correct_body(client,
```
This test returns `400` when it's run together with other tests, because of the parser. Now we create one parser instance for all routes within a file. And after that each route adds each own arguments to the parser.

This means that we run all tests for routes within a file the parser is configured by two routes and has arguments for two routes, but we need a parser to have arguments only for one route each time.

This should be fixed by creating a separate parser instance for each route. Created a [ticket](https://github.com/fenya123/forum123/issues/154) for that.

EugeneTurkin committed:
149: Add tests for api/posts endpoints

We want to cover our REST API part of the codebase with unit-tests.

In the scope of this task we need to cover code which is working with posts in our REST API.

birthdaysgift reviewed (changes requested):
Need to add test for this case:
![image](https://user-images.githubusercontent.com/35929293/236688176-b202e3ca-26bd-4d0f-8d0d-e07a45a88d0c.png)
